File: starterAIs/Starter.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Player
{
    static void Main(string[] args)
    {
        string[] inputs;
        int numberOfCells = int.Parse(Console.ReadLine()); // amount of hexagonal cells in this map
        var cells = new List<Cell>();
        for (int i = 0; i < numberOfCells; i++)
        {
            inputs = Console.ReadLine().Split(' ');
            var cell = new Cell
            {
                Index = i,
                Type = (CellType)int.Parse(inputs[0]), // 0 for empty, 1 for eggs, 2 for food
                Resource = int.Parse(inputs[1]) // the initial amount of eggs/crystals on this cell
            };
            cell.AddNeighbor(int.Parse(inputs[2]), int.Parse(inputs[3]), int.Parse(inputs[4]), int.Parse(inputs[5]),
                int.Parse(inputs[6]), int.Parse(inputs[7])); // the index of the neighbouring cell for each direction
            cells.Add(cell);
        }

        int numberOfBases = int.Parse(Console.ReadLine());
        var myBaseIndices = Console.ReadLine().Split(' ').Take(numberOfBases).Select(int.Parse);
        var oppBaseIndices = Console.ReadLine().Split(' ').Take(numberOfBases).Select(int.Parse);

        // game loop
        while (true)
        {
            for (int i = 0; i < numberOfCells; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                cells[i].Resource = int.Parse(inputs[0]); // the current amount of eggs/crystals on this cell
                cells[i].MyAnts = int.Parse(inputs[1]); // the amount of your ants on this cell
                cells[i].OppAnts = int.Parse(inputs[2]); // the amount of opponent ants on this cell
            }

            var actions = new List<string>();
            // Write an action using Console.WriteLine()
            // To debug: Console.Error.WriteLine("Debug messages...");

            // From here, add the actions you want to the list, e.g.
            // actions.add("LINE 1 5 1")
            // actions.add("BEACON 1 5")

            // WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
            if (!actions.Any())
            {
                Console.WriteLine("WAIT");
            }
            else
            {
                // Print all actions
                Console.WriteLine(string.Join(';', actions));
            }
        }
    }
}

enum CellType
{
    Empty = 0,
    Egg = 1,
    Crystal = 2
}

class Cell
{
    public int Index { get; set; }
    public CellType Type { get; set; }
    public int Resource { get; set; }
    public IList<int> Neighbors { get; set; } = new List<int>();
    public int MyAnts { get; set; }
    public int OppAnts { get; set; }

    public void AddNeighbor(params int[] neighborIndices)
    {
        foreach (var neighborIndex in neighborIndices.Where(n => n > -1))
        {
            Neighbors.Add(neighborIndex);
        }
    }
}

File: starterAIs/Starter.java
import java.util.*;
import java.io.*;
import java.math.*;


class Cell {
    int index;
    int cellType;
    int resources;
    List<Integer> neighbors;
    int myAnts;
    int oppAnts;

    public Cell(int index, int cellType, int resources, List<Integer> neighbors, int myAnts, int oppAnts) {
        this.index = index;
        this.cellType = cellType;
        this.resources = resources;
        this.neighbors = neighbors;
        this.myAnts = myAnts;
        this.oppAnts = oppAnts;
    }
}

class Player {

    public static void main(String args[]) {

        List<Cell> cells = new ArrayList<>();

        Scanner in = new Scanner(System.in);
        int numberOfCells = in.nextInt(); // amount of hexagonal cells in this map
        for (int i = 0; i < numberOfCells; i++) {
            int cellType = in.nextInt(); // 0 for empty, 1 for eggs, 2 for crystal
            int initialResources = in.nextInt(); // the initial amount of eggs/crystals on this cell
            List<Integer> neighbors = new ArrayList<>(); // the index of the neighbouring cell for each direction
            neighbors.add(in.nextInt());
            neighbors.add(in.nextInt());
            neighbors.add(in.nextInt());
            neighbors.add(in.nextInt());
            neighbors.add(in.nextInt());
            neighbors.add(in.nextInt());
            Cell cell = new Cell(i, cellType, initialResources, neighbors, 0, 0);
            cells.add(cell);
        }
        int numberOfBases = in.nextInt();
        List<Integer> myBases = new ArrayList<>();
        List<Integer> oppBases = new ArrayList<>();

        for (int i = 0; i < numberOfBases; i++) {
            int myBaseIndex = in.nextInt();
            myBases.add(myBaseIndex);
        }
        for (int i = 0; i < numberOfBases; i++) {
            int oppBaseIndex = in.nextInt();
            oppBases.add(oppBaseIndex);
        }

        // game loop
        while (true) {
            for (int i = 0; i < numberOfCells; i++) {
                int resources = in.nextInt(); // the current amount of eggs/crystals on this cell
                int myAnts = in.nextInt(); // the amount of your ants on this cell
                int oppAnts = in.nextInt(); // the amount of opponent ants on this cell

                cells.get(i).resources = resources;
                cells.get(i).myAnts = myAnts;
                cells.get(i).oppAnts = oppAnts;
            }

            // Write an action using System.out.println()
            // To debug: System.err.println("Debug messages...");

            List<String> actions = new ArrayList<>();
            
            //TODO: choose actions to perform and push them into actions. E.g:
            for (Cell cell : cells) {
                if (cell.resources > 0) {
                    actions.add("LINE " + myBases.get(0) + " " + cell.index + " 1");
                    break;
                }
            }

            if (actions.size() == 0) {
                System.out.println("WAIT");
            } else {
                // WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
                System.out.println(String.join(";", actions));
            }
        }
    }
}